examples/pettingzoo/meltingpot_env.py

```diff
@@ -0,0 +1,149 @@
+  elif isinstance(spec, dict):
+    return spaces.Dict({key: _spec_to_space(s) for key, s in spec.items()})
+  else:
+    raise ValueError('Unexpected spec: {}'.format(spec))
+
+def parallel_env(env_config, max_cycles=MAX_CYCLES):
+    return _parallel_env(env_config, max_cycles)
+
+def raw_env(env_config, max_cycles=MAX_CYCLES):
+    return parallel_to_aec_wrapper(parallel_env(env_config, max_cycles))
+
+def env(env_config, max_cycles=MAX_CYCLES):
+    aec_env = raw_env(env_config, max_cycles)
+    aec_env = wrappers.AssertOutOfBoundsWrapper(aec_env)
+    aec_env = wrappers.OrderEnforcingWrapper(aec_env)
+    return aec_env
+
+class MeltingPotPettingZooEnv(ParallelEnv):
+  """An adapter between the Melting Pot substrates and PettingZoo's ParallelEnv"""
+
+  def __init__(self, env_config, max_cycles):
+    self.env_config = config_dict.ConfigDict(env_config)
+    self.max_cycles = max_cycles
+    self._env = substrate.build(self.env_config)
+    self._num_players = len(self._env.observation_spec())
+    self.possible_agents = [
+        PLAYER_STR_FORMAT.format(index=index)
+        for index in range(self._num_players)
+    ]
+    observation_space = _remove_world_observations_from_space(
+        _spec_to_space(self._env.observation_spec()[0]))
+    self.observation_space = lru_cache(maxsize=None)(lambda agent_id: observation_space)
+    action_space = _spec_to_space(self._env.action_spec()[0])
+    self.action_space = lru_cache(maxsize=None)(lambda agent_id: action_space)
+    self.state_space = _spec_to_space(self._env.observation_spec()[0]['WORLD.RGB'])
+
+  def state(self):
+    return self._env.observation()
+
+  def reset(self):
+    """See base class."""
+    timestep = self._env.reset()
+    self.agents = self.possible_agents[:]
+    self.num_cycles = 0
+    return _timestep_to_observations(timestep)
+
+  def step(self, action):
+    """See base class."""
+    actions = [action[agent] for agent in self.agents]
+    timestep = self._env.step(actions)
+    rewards = {
+        agent: timestep.reward[index]
+        for index, agent in enumerate(self.agents)
+    }
+    self.num_cycles += 1
+    done = timestep.last() or self.num_cycles >= self.max_cycles
+    dones = {agent: done for agent in self.agents}
+    infos = {agent: {} for agent in self.agents}
+    if done:
+        self.agents = []
+
+    observations = _timestep_to_observations(timestep)
+    return observations, rewards, dones, infos
+
+  def close(self):
+    """See base class."""
+    self._env.close()
+  
+  def seed(self, seed=None):
+    raise NotImplementedError
+  
+  def render(self, mode="human", filename=None):
+    rgb_arr = self.state()['WORLD.RGB']
+    if mode == "human":
+        plt.cla()
+        plt.imshow(rgb_arr, interpolation="nearest")
+        if filename is None:
+            plt.show(block=False)
+        else:
+            plt.savefig(filename)
+        return None
+    return rgb_arr
+
+class _parallel_env(MeltingPotPettingZooEnv, EzPickle):
```

Comment:
Class names should be CamelCase starting with capitals. If the class is private to the module, it's OK to prefix with an underscore.